VocaDbWeb/Views/Shared/Partials/_LayoutScripts.cshtml

```diff
@@ -15,13 +17,23 @@
 
 <script type="text/javascript">
 	vdb = vdb || {};
-	vdb.values = vdb.values || {};
-	vdb.values.baseAddress = '@RootPath';
-	vdb.values.languagePreference = @((int)Login.Manager.LanguagePreference);
-	vdb.values.isLoggedIn = @ToJS(UserContext.IsLoggedIn);
-	vdb.values.loggedUserId = @ToJS(UserContext.LoggedUserId);
-	vdb.values.culture = @ToJS(Culture);
-	vdb.values.uiCulture = @ToJS(UICulture);
+	vdb.values = @ToJS(new
+	{
+		LockdownMessage = AppConfig.LockdownMessage,
+		BaseAddress = RootPath,
+		LanguagePreference = (int)Login.Manager.LanguagePreference,
+		IsLoggedIn = UserContext.IsLoggedIn,
+		LoggedUserId = UserContext.LoggedUserId,
+		Culture = Culture,
+		UICulture = UICulture,
+		LoggedUser = Login.User is ServerOnlyUserWithPermissionsContract loggedUser ? new
+		{
+			Id = loggedUser.Id,
+			Name = loggedUser.Name,
+			Active = loggedUser.Active,
+			EffectivePermissions = loggedUser.EffectivePermissions.Select(p => p.Id).ToArray(),
+		} : null,
+	});
 	vdb.resources = vdb.resources || {};
 
 	var model = {

```

Comment:
@riipah Is it okay to send effective permissions back to the client?

Example:

```json
{
    "baseAddress": "/",
    "languagePreference": 0,
    "loggedUserId": 39,
    "culture": "en",
    "uiCulture": "en",
    "loggedUser": {
        "id": 39,
        "name": "Miku",
        "active": true,
        "effectivePermissions": [
            "4f79b01a-7154-4a7f-bc87-a8a9259a9905",
            "be2deee9-ee12-48b4-a9a5-e369915fc156",
            "d762d720-79ef-4e60-8397-1d638c26d82b",
            "2ab6da19-bc5a-4a28-86d4-8ddce399ca0b",
            "85c730a1-402c-4922-b617-6c6068f5f98e",
            "cf39509b-b9c5-4efc-9b13-2743ffec9aac",
        ]
    }
}
```

Reply:
I don't see a problem with sending user's permission tokens to the client. For example, MusicBrainz sends a sanitized version of [Catalyst](https://en.wikipedia.org/wiki/Catalyst_(software)) context to the client, including editor's [privileges](https://github.com/metabrainz/musicbrainz-server/blob/f7d1d109e51b9c0313de3bdd64c1ce0f543e73f1/root/utility/activeSanitizedEditor.js#L26). In MusicBrainz, privileges are represented as a [bitfield](https://github.com/metabrainz/musicbrainz-server/blob/f7d1d109e51b9c0313de3bdd64c1ce0f543e73f1/root/constants.js#L59-L70).

Reply:
It's okay. It's not secret information (for the user themselves). Just make sure the client can't do anything with those permissions without first checking with the server.

Reply:
I see, thanks!